File: from-2401-to-2500/2441-largest-positive-integer-that-exists-with-its-negative/solution.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        nums.sort()
        l, r = 0, len(nums) - 1
        while l < r and nums[l] < 0 and nums[r] > 0:
            if -nums[l] == nums[r]:
                return nums[r]
            elif -nums[l] > nums[r]:
                l += 1
            else:
                r -= 1
                
        return -1

File: from-2401-to-2500/2441-largest-positive-integer-that-exists-with-its-negative/solution.go
func findMaxK(nums []int) int {
    hash := make(map[int]bool)
    ans := 0
    for _, num := range nums {
        if hash[-num] && abs(num) > ans {
            ans = abs(num)
        }
        hash[num] = true
    }
    if ans == 0 {
        return -1
    }
    return ans
}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}